frontend/src/metabase/visualizations/components/ScalarValue.styled.tsx

```diff
@@ -11,3 +22,94 @@ export const ScalarRoot = styled.div`
   width: 100%;
   height: 100%;
 `;
+
+export const ScalarValueWrapper = styled.h1<ScalarValueProps>`
+  cursor: pointer;
+  &:hover {
+    color: ${color("brand")};
+  }
+
+  font-size: ${props => {
+    const MIN_SIZE_REM = 2.2;
+    const MAX_SIZE_REM = 12;
+
+    const fontSizeAdjustment = computeFontSizeAdjustment(props);
+
+    // fail-safe to make sure the font size is within a reasonable range
+    const clampedScaledFontSize = Math.min(
+      Math.max(MIN_SIZE_REM + fontSizeAdjustment, MIN_SIZE_REM),
+      MAX_SIZE_REM,
+    );
+
+    return `${clampedScaledFontSize}rem`;
+  }};
+`;
+
+export const WIDTH_ADJUSTMENT_FACTOR = 4;
+export const HEIGHT_ADJUSTMENT_FACTOR = 4;
+export const MAX_HEIGHT_GRID_SIZE = 10;
+
+export function computeFontSizeAdjustment({
```

Comment:
Could we move this function to a separate file or to utils? I don't think it belongs to a `styled.tsx` file